packages/framework/attributor/src/mixinAttributor.ts

```diff
@@ -187,159 +106,33 @@ export const mixinAttributor = (
 				containerRuntimeCtor,
 				// eslint-disable-next-line @typescript-eslint/no-explicit-any
 			} as any)) as ContainerRuntimeWithAttributor;
-			runtime.runtimeAttributor = runtimeAttributor as RuntimeAttributor;
-
-			const logger = createChildLogger({
-				logger: runtime.baseLogger,
-				namespace: "Attributor",
-			});
 
-			// Note: this fetches attribution blobs relatively eagerly in the load flow; we may want to optimize
-			// this to avoid blocking on such information until application actually requests some op-based attribution
-			// info or we need to summarize. All that really needs to happen immediately is to start recording
-			// op seq# -> attributionInfo for new ops.
-			await PerformanceEvent.timedExecAsync(
-				logger,
-				{
-					eventName: "initialize",
-				},
-				async (event) => {
-					await runtime.runtimeAttributor?.initialize(
-						deltaManager,
-						quorum,
-						baseSnapshot,
-						async (id) => runtime.storage.readBlob(id),
-						shouldTrackAttribution,
+			let runtimeAttributor: RuntimeAttributorDataStoreChannel | undefined;
+			if (shouldTrackAttribution) {
+				if (existing) {
+					const entryPoint = await runtime.getAliasedDataStoreEntryPoint(
+						attributorDataStoreAlias,
 					);
-					event.end({
-						attributionEnabledInConfig: shouldTrackAttribution,
-						attributionEnabledInDoc: runtime.runtimeAttributor
-							? runtime.runtimeAttributor.isEnabled
-							: false,
-					});
-				},
-			);
+					runtimeAttributor = (await entryPoint?.get()) as
+						| RuntimeAttributorDataStoreChannel
+						| undefined;
+				} else {
+					const datastore = await runtime.createDataStore(RuntimeAttributorFactory.type);
+					const result = await datastore.trySetAlias(attributorDataStoreAlias);
+					assert(result === "Success", "Failed to set alias for attributor data store");
+					runtimeAttributor =
+						(await datastore.entryPoint.get()) as RuntimeAttributorDataStoreChannel;
+					assert(runtimeAttributor !== undefined, "Attributor should be defined");
+				}
+				runtime._runtimeAttributor = runtimeAttributor?.IRuntimeAttributor;
+			}
 
 			return runtime;
 		}
 
-		private runtimeAttributor: RuntimeAttributor | undefined;
+		private _runtimeAttributor: IRuntimeAttributor | undefined;
 
-		protected addContainerStateToSummary(
-			summaryTree: ISummaryTreeWithStats,
-			fullTree: boolean,
-			trackState: boolean,
-			telemetryContext?: ITelemetryContext,
-		): void {
-			super.addContainerStateToSummary(summaryTree, fullTree, trackState, telemetryContext);
```

Comment:
Is there some back-compat consideration needed here?  Since now the summary will be down alongside the other DataStores, rather than in a custom top-level blob, right?

Reply:
No, this is just experimental package, I think I can make this change without back compat consideration as it also not used by loop.